pypeit/reduce.py

```diff
@@ -1122,9 +1129,10 @@ def find_objects_pypeline(self, image, std_trace=None,
             else:
                 sig_thresh = self.par['reduce']['findobj']['sig_thresh']
 
-            # TODO we need to add QA paths and QA hooks. QA should be
-            # done through objfind where all the relevant information
-            # is. This will be a png file(s) per slit.
+            # dict for object finding QA
+            objfindQA_dict = {'SAVE_TO_DISK': save_objfindQA, 'BASENAME': self.basename, 'SLITORD_ID': slit_spat,
+                              'OUTDIR': os.path.join(self.par['rdx']['redux_path'], self.par['rdx']['qadir']),
+                              'NEG': neg}
 
```

Comment:
Run the code to synthesise the filename here ,and then only pass the core method the filename. This has the benefit that you reduce dependencies and failure modes in the core objfind method. For example, a string cannot fail, but a call to qa.set_qa_filename can fail, and that dependency is completely unnecessary since you can generate the filename outside of that routine.

Reply:
Done.